sdk/storage/AzureStorageBlob/Source/DataStructures/URLSessionTransferManager.swift

```diff
@@ -0,0 +1,389 @@
+        if let operation = transfer.operation {
+            operationQueue.pause(operation)
+        }
+    }
+
+    internal func pause(transfer: BlobTransfer) {
+        // Cancel the operation and any associated block operations
+        if let operation = transfer.operation {
+            operationQueue.remove(operation)
+            // transfer.state = .paused
+            for block in transfer.transfers {
+                block.state = .paused
+                if let blockOp = block.operation {
+                    operationQueue.remove(blockOp)
+                }
+            }
+        }
+    }
+
+    // MARK: Resume Operations
+
+    public func resume(transfer: Transfer) {
+        guard reachability?.isReachable ?? false else { return }
+        let allowed: [TransferState] = [.paused]
+        guard allowed.contains(transfer.state) else { return }
+        transfer.state = .pending
+
+        switch transfer {
+        case let transfer as BlockTransfer:
+            operationQueue.add(BlockOperation(withTransfer: transfer))
+        case let transfer as BlobTransfer:
+            for blockTransfer in transfer.transfers {
+                if allowed.contains(blockTransfer.state) {
+                    blockTransfer.state = .pending
+                }
+            }
+            queueOperationsFor(blobTransfer: transfer)
+        default:
+            fatalError("Unrecognized transfer type: \(transfer.self)")
+        }
+        operation(transfer.operation, didChangeState: transfer.state)
+    }
+
+    // MARK: Core Data Operations
+
+    public func loadContext() {
+        // Hydrate operationQueue from CoreData
+        guard let context = persistentContainer?.viewContext else {
+            fatalError("Unable to load persistent container.")
+        }
+        context.mergePolicy = NSMergePolicy.mergeByPropertyObjectTrump
+
+        let predicate = NSPredicate(format: "parent = nil")
+        let blockRequest: NSFetchRequest<BlockTransfer> = BlockTransfer.fetchRequest()
+        blockRequest.predicate = predicate
+        if let results = try? context.fetch(blockRequest) {
+            for transfer in results {
+                add(transfer: transfer)
+            }
+        }
+        let blobRequest: NSFetchRequest<BlobTransfer> = BlobTransfer.fetchRequest()
+        blobRequest.predicate = predicate
+        if let results = try? context.fetch(blobRequest) {
+            for transfer in results {
+                print(transfer.debugString)
+                add(transfer: transfer)
+            }
+        }
+    }
+
+    public func saveContext() {
+        DispatchQueue.main.async { [weak self] in
+            guard let context = self?.persistentContainer?.viewContext else { return }
+            if context.hasChanges {
+                do {
+                    try context.save()
+                } catch {
+                    // Replace this implementation with code to handle the error appropriately.
```

Comment:
Same as above - what should we actually do here? Is this a TODO? if so we should mark with a //TODO. It's written like they're instructions for the end user.

Reply:
I removed both of these boilerplate comments. I will address removing all of the 20 or so fatalError occurrences once I have upload working with TM since there's no good reason to have fatalError in production code.